style.md

```diff
@@ -1894,6 +1896,172 @@ this contract. Specifying field names inside tags makes the contract explicit,
 and it guards against accidentally breaking the contract by refactoring or
 renaming fields.
 
+### Don't fire-and-forget goroutines
```

Comment:
Given the sentence toward the end of the change:

> Spawns the worker only if the user requests it.

it might be worth directly stating that goroutines should not be started implicitly, but rather as a result of function call that could reasonably be expected to launch goroutine(s). (Ideally we could refer folks to [POLA](https://en.wikipedia.org/wiki/Principle_of_least_astonishment), but that's a broader topic than this change.)

Reply:
I thought about this, but I'm not sure where to incorporate this.
It could be argued that if the returned object has a Close/Stop/Shutdown method, the goroutine is an implementation detail.
The contract is just that if you don't call the method when you're done, you're leaking resources.

Reply:
Yeah, fair. We can address all of this later (if ever).